pkg/unarchive/dmg.go

```diff
@@ -0,0 +1,132 @@
+package unarchive
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/aquaproj/aqua/pkg/exec"
+	"github.com/schollz/progressbar/v3"
+	"github.com/spf13/afero"
+)
+
+const FormatDMG string = "dmg"
+
+type dmgUnarchiver struct {
+	dest string
+}
+
+func cpFile(fs afero.Fs, src, dst string) error {
+	srcFile, err := fs.Open(src)
+	if err != nil {
+		return fmt.Errorf("open a file: %w", err)
+	}
+	defer srcFile.Close()
+
+	dstFile, err := fs.Create(dst)
+	if err != nil {
+		return fmt.Errorf("create a file: %w", err)
+	}
+	defer dstFile.Close()
+
+	if _, err = io.Copy(dstFile, srcFile); err != nil {
+		return fmt.Errorf("copy a file: %w", err)
+	}
+
+	return nil
+}
+
+func cpDir(fs afero.Fs, src, dst string) error {
+	fileInfos, err := afero.ReadDir(fs, src)
+	if err != nil {
+		return fmt.Errorf("read a directory: %w", err)
+	}
+
+	for _, fileInfo := range fileInfos {
+		srcPath := filepath.Join(src, fileInfo.Name())
+		dstPath := filepath.Join(dst, fileInfo.Name())
+
+		if fileInfo.IsDir() {
+			if err := fs.MkdirAll(dstPath, dirPermission); err != nil {
+				return fmt.Errorf("create a directory: %w", err)
+			}
+			if err := cpDir(fs, srcPath, dstPath); err != nil {
+				return fmt.Errorf("copy a directory: %w", err)
+			}
+		} else {
+			if err := cpFile(fs, srcPath, dstPath); err != nil {
+				return fmt.Errorf("copy a file: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func writeDmgFile(m io.Writer, body io.Reader, dest string) error {
```

Comment:
Is this function necessary?
I think we can simply use `io.Copy` instead.

Reply:
When I tried purely using io.Copy, there was no particular problem...

I was writing the code by fumbling, so I didn't think about writing it more simply.

https://github.com/aquaproj/aqua/pull/1798/commits/d8546e9ab06f925b68c0c3f9bb566986005340b4